### ISSUE TITLE
Cycle segment showing as walking

### ISSUE THREAD

**Issue by HilaryN** [cyclestreets/android#497]
e.g. in the following journey:
https://cycle.st/j78841326 - Lennard Road to Alley from Park Road
Park Road is showing as a walking segment when it should be cycling.
On the website it shows a a cycling segment.
It may be something to do with 2 (short?) segments being combined.

**Comment by HilaryN** [comment]
In fact this is happening when 2 sections are combined, because one of them is very short.  
If one of the sections being combined is a walking section then the whole combined section will show as walking.

**Comment by mvl22** [comment]
Is the API giving the correct data? I.e. is this an app rendering issue?

**Comment by oliverlockwood** [comment]
I'm pretty sure the API is not going to be at fault here.

For a reminder of context: The segment-combining is very useful, because it was extraordinarily irritating (and at times dangerous) to e.g turn right as instructed at a staggered T-crossroads, and be well past the junction by the time the voice instruction says something like "Continue 5 metres. Get ready to turn left". So that's why we have instructions like "Get ready to Turn right then turn left" instead.

When I wrote the segment-combining code, I simply hadn't thought of the possibility of one of the segments being dismounted and the other not.  Probably because personally I tend to prefer routes that don't involve walking, as I'd rather just stay on the bike!

The easiest fix would probably be to introduce a check and, if the mounted/dismounted states of the two segments are not identical, then don't do a proposed segment-combining operation.

But I'd be interested in the thoughts of others, maybe someone has a better idea?

**Comment by HilaryN** [comment]
Yes, it's definitely not the API.  It's happening within the app when segments are combined (melded), and I agree this is a neat feature, which is why I'm hesitant to meddle with it.

I guess if the first (less-than-20m) segment is walking and the second isn't, we don't need to do the segment-melding, because there is less chance they will miss the turn if they are walking the first bit.

If it's the other way round then we can meld them - it's not such an issue if the short segment mistakenly shows as walking.
I'm still mulling it over - I was thinking about keeping the segments separate and just putting the combined instruction into the first segment, but then it'll have the wrong street name.  (Maybe I could fix that. too.)

I'll also need to look into the roundabout-melding and the bridge-melding as they do the same thing.
Linking to the previous issue here as there are some examples in there which should come in handy for testing:
https://github.com/cyclestreets/android/issues/276

I also prefer routes which don't have too much walking and it was a bit off-putting seeing the long section (incorrectly) marked as walking.

**Comment by HilaryN** [comment]
So on the website it shows as separate segments:

<img width="801" alt="CycleStreets website" src="https://user-images.githubusercontent.com/59799824/187054692-3f9df34d-e463-4db9-869a-ecb439ab9b1f.PNG">
Currently on the app it shows as a melded segment:

<img width="522" alt="Current Android" src="https://user-images.githubusercontent.com/59799824/187054699-068b6dd2-65bc-40e8-a838-e02bbf3c45e9.PNG">
If one segment is walk and one is cycle, I could leave the segments separate but put the combined instruction into the first segment with the second segment's street name and leave the second segment as it is so that they show correctly on the map:

<img width="524" alt="Proposed android" src="https://user-images.githubusercontent.com/59799824/187054739-093651dc-6a5a-4136-aefd-994a376b0209.PNG">

**Comment by HilaryN** [comment]
The roundabout melding shouldn't be an issue.  The Bridge melding could be, but I can't find a real example of mixed walking/cycling so I'll have to fudge the test.

**Comment by HilaryN** [comment]
Left to right: Bridge on website; melded bridge on current version of Android app; split out bridge instructions if the 2 segments were to be a mix of walking / cycling (I'll leave it melded if they are both cycling - this is fudged just to illustrate what it would look like):

![image](https://user-images.githubusercontent.com/59799824/187766195-f1ba4b5f-92e7-4bd1-8e3a-32f241ae89b8.png)

The first screenshot here shows the melded bridge and the other 2 show how it would look if the 2 segments were split out (again, fudged - they will remain melded if both segments are cycling or both walking):

![image](https://user-images.githubusercontent.com/59799824/187768013-250af219-7f86-49a6-a731-9d923286485b.png)

**Comment by mvl22** [comment]
Surely what should happen is that the visual map should never be melded, but the voice instructions may be.

So logically the melding should logically be moved up a level into the voice data logic and out of any central processing, as melding is completely specific to voice. Am I right?

**Comment by oliverlockwood** [comment]
@mvl22 No, I strongly disagree with you here.

Melding is deliberately part of the rendering of the route in terms of directional arrows and visual instructions also.

For example, from @HilaryN's screenshots above: 
![Screenshot 2022-09-05 at 13 51 06](https://user-images.githubusercontent.com/6017680/188453592-0f72fe50-65e0-49ac-ab28-d3e2e772fe1c.png)

**Comment by mvl22** [comment]
> For example, from @HilaryN's screenshots above:

Yes, I agree that the directions list should also do that. Sorry, I should have been clearer. I regard that as linked to the voice side.

All I'm saying is that the map itself should always show individual segments exactly as they are from the API.
